bots/engine.py

```diff
@@ -85,17 +85,19 @@ def start():
         botsglobal.logger = botsinit.initenginelogging(process_name)
         atexit.register(logging.shutdown)
     except Exception as msg:
+        print('Error initialising logging:' + msg)
         botslib.sendbotserrorreport('[Bots severe error] Bots is not running anymore','Bots does not run because logging is not possible.\nOften a rights problem.\n')
         sys.exit(1)
     else:
-        if botsglobal.ini.get('settings','log_file_number','') != 'daily':
+        if botsglobal.ini.get('settings','log_when',None) != 'daily':
```

Comment:
not sure I understand this. there is no 'log_when'. if new, what does it do?
it would be good to have an option to log engine to one log file (per day), as you have suggested..

Reply:
I thought you had added log_when? It is there in botsinit so I changed engine to use it also.
Probably changed because botsglobal.ini.getint('settings','log_file_number',10) gives an error because 'daily' is not an int.
Also means you can set daily AND number of days to keep.

Reply:
I originally made this change years ago and probably posted to the mailing list, but only used log_file_number at that time. In my installations I keep daily logs and do daily reporting on those in botsengine postcleanup.

Reply:
Needs to be added in bots.ini but comment out by default

#log_file_number: number of rotating log files. Value: number. Default: 10
log_file_number = 10
#for one merged log per day, set log_when = daily. Default: Each run uses it's own log file
#log_when = daily

Reply:
could also use a setting in bots.ini to explicitly state the default; it will not affect operation of this setting.

#for one merged log per day, set log_when = daily. Default: Each run uses it's own log file
log_when = each run